### PR TITLE
Bump the version from Xenial to Bionic acceptance tests

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following PR ensures that we update the acceptance tests
to ensure that xenial is now bionic. 

----

This makes sure that all LXD images by default are Bionic for 2.5
release. The issue with this is that it can cause issues from this,
because we're now bumping the LTS version for EVERYTHING.

A lot of testing should be done to ensure that we don't cause other
issues for other parts of the system.

## QA steps

Test that launching LXD providers is possible AND that other 
providers/machines aren't broken in changing this.

## Documentation changes

The default LTS for juju is changing from Xenial to Bionic. You can still
over ride this using the series flag if required.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1785124

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/d6dMgDud/439-bug-lxd-controller-running-xenial-not-bionic

**Comment by SimonRichardson** [comment]
I guess I didn't communicate the request of the review, very well and that's my mistake. I was wondering if you might give some idea as to what the fallout of this change would be (other than tests)? Are we likely hit a mongo issue, as I know there are some issues around mongo versions between series...?

**Comment by anastasiamac** [comment]
Oh I see :)
To be honest, I would hope that there would not be any "fallout". We have been testing with bionic for a bit know. Of course, we have been testing by explicitly asking for bionic series...whereas now we want it to be picked up by default, but our logic is kind of intrinsically not different ...

So besides, the unit tests that would need to be changed to check for latest, default LTS to be 'bionic', the manual testing would need to ensure that:
*  your default bootstrap, without specifying series, stands up bionic. This will ensure that image and agent binary selection used the code changes in this PR.
* you default deployment of a charm, without specifying a series, stands up a bionic unit machine. This will test image and agent binary default series selection as well as the charm series selection - you will need to have charm that is 'bionic' savy.

I am fairly certain we have unit and CI tests to cover it but it will also give you a peace of mind before this PR lands :)

If you do encounter anything that is a 'fallout', it might be worthwhile sharing the experience on Discource for future reference when next LTS rolls out?

**Comment by SimonRichardson** [comment]
Some of the tests rely on setup logic in `export_test.go` files. I feel they shouldn't be in there, as `export_test.go` files are used specifically for re-exporting types/functions to help with unit testing and isn't the normal place for these sorts of things to reside. 

What's the thought of changing this so future people will more likely find it?

**Comment by SimonRichardson** [comment]
@Veebers I've updated what I think are good updates to the acceptance tests, I could be wrong, but could you do me a favour and take a look?

**Comment by SimonRichardson** [comment]
!!build!! 

mongo session failure caused the bot check error.

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
I've updated this PR to only be acceptance tests and another PR for the go code, so that we can merge and unblock the go code.

File: acceptancetests/repository/bundles-kubernetes-core-lxd.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
-series: xenial
+series: bionic
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply acceptancetests/repository/bundles-kubernetes-core-lxd.yaml]: I've updated these in https://github.com/juju/juju/pull/9023/commits/f8ed817f781b35d77f7292980f355a9ddd5c6d82

**Pull request merged**